otherlibs/win32unix/channels.c

```diff
@@ -23,6 +23,42 @@
 #include <fcntl.h>
 #include <io.h>
 
+/* Check that the given file descriptor has "stream semantics" and
+   can therefore be used as part of buffered I/O.  Things that
+   don't have "stream semantics" include block devices and
+   UDP (datagram) sockets.
+   Returns 0 if OK, a Win32 error code if error. */
+
+static DWORD win_check_stream_semantics(value handle)
+{
+  switch (Descr_kind_val(handle)) {
+  case KIND_HANDLE:
+    switch (GetFileType(Handle_val(handle)) & ~FILE_TYPE_REMOTE) {
+    case FILE_TYPE_DISK: case FILE_TYPE_CHAR: case FILE_TYPE_PIPE:
+      return 0;
+    default: {
+      DWORD err = GetLastError();
+      return err == 0 ? ERROR_INVALID_ACCESS : err;
```

Comment:
I would write `NO_ERROR` here, which is slightly clearer than `0`

Reply:
Nice suggestion, implemented in 0724940